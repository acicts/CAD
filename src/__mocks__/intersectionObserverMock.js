global.IntersectionObserver = class IntersectionObserver {
	constructor() {}

	observe() {
		return null;
	}

	disconnect() {
		return null;
	}

	unobserve() {
		return null;
	}
};
