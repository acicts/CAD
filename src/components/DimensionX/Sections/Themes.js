import React, { useState, useEffect } from "react";
import SectionTopic from "../../DimensionX/SectionTopic";
import SubSectionTopic from "../../CodeFLow/SubSectionTopic";
import classes from "../../../styles/DimensionX/Themes.module.css";

let init = false;

const Themes = () => {
	const calculateTimeLeft = () => {
		let year = new Date().getFullYear();
		let difference = +new Date(2022, 2, 1) - +new Date();
		let timeLeft = {};

		if (difference > 0) {
			timeLeft = {
				days: Math.floor(difference / (1000 * 60 * 60 * 24)),
				hours: Math.floor((difference / (1000 * 60 * 60)) % 23),
				minutes: Math.floor((difference / 1000 / 60) % 60),
				seconds: Math.floor((difference / 1000) % 60),
			};
		}

		return timeLeft;
	};

	const textColor = "#07253a";
	const pendulamColor = "#07253a";
	const [time, setTime] = useState(null);
	const [mins, setMins] = useState(null);
	const [hours, setHours] = useState(null);
	const [days, setDays] = useState(null);
	const [eventStarted, setEventStarted] = useState(false);
	useEffect(() => {
		const timeLeft = calculateTimeLeft();
		if (time === null) {
			if (timeLeft.seconds >= 0) {
				setTime(timeLeft.seconds);
				setHours(timeLeft.hours);
				setMins(timeLeft.minutes);
				setDays(timeLeft.days);
			} else {
				setEventStarted(true);
			}
		} else if (!eventStarted) {
			const timer = setTimeout(() => {
				if (days === 0 && hours === 0 && mins === 0 && time === 0) {
					setEventStarted(true);
				} else {
					if (hours === 0) {
						if (days > 0) {
							setDays((curr) => curr - 1);
							setHours(24);
						}
					}
					if (mins === 0) {
						if (hours > 0) {
							setHours((curr) => curr - 1);
							setMins(60);
						}
					}
					if (time === 0) {
						if (mins > 0) {
							setMins((curr) => curr - 1);
							setTime(60);
						}
					} else {
						setTime((curr) => curr - 1);
					}
				}
			}, 1000);

			return () => clearTimeout(timer);
		}
	}, [eventStarted, days, hours, mins, time]);
	return (
		<section className={classes.Container} id="themes">
			<SectionTopic
				title={"Themes"}
				color={textColor}
				pendulamColor={"#46dd9a"}
			>
				{!eventStarted ? (
					<div className={classes.TimeContainer}>
						<h1>Themes Will Be Revealed In!</h1>
						<div className={classes.TimeBar}>
							<div>
								<p>{days}</p>
								<p>Days</p>
							</div>
							<div className={classes.Dots}>:</div>
							<div>
								<p>{hours}</p>
								<p>Hours</p>
							</div>
							<div className={classes.Dots}>:</div>
							<div>
								<p>{mins}</p>
								<p>Mins</p>
							</div>
							<div className={classes.Dots}>:</div>
							<div>
								<p>{time}</p>
								<p>Seconds</p>
							</div>
						</div>
					</div>
				) : (
					<div className={classes.SubContainer}>
						<SubSectionTopic
							title={"Senior Category"}
							color={textColor}
							pendulamColor={pendulamColor}
						>
							<ul>
								<li>Senior Theme ABCDE</li>
								<li>Senior Theme ABCDE</li>
								<li>Senior Theme ABCDE</li>
								<li>Senior Theme ABCDE</li>
								<li>Senior Theme ABCDE</li>
							</ul>
						</SubSectionTopic>
						<SubSectionTopic
							title={"Junior Category"}
							color={textColor}
							pendulamColor={pendulamColor}
						>
							<ul>
								<li>Junior Theme ABCDE</li>
								<li>Junior Theme ABCDE</li>
								<li>Junior Theme ABCDE</li>
							</ul>
						</SubSectionTopic>
					</div>
				)}
			</SectionTopic>
		</section>
	);
};

export default Themes;
