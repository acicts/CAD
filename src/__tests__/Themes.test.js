import React from "react";
import Themes from "../components/CodeFLow/sections/Themes";
import { render } from "@testing-library/react";
import "../__mocks__/intersectionObserverMock";

afterEach(() => {
	jest.useRealTimers();
});

test("Should render themes after time is 0", () => {
	jest.useFakeTimers().setSystemTime(new Date(2022, 2, 1));
	const { queryByTestId } = render(<Themes />);
	const string = "senior-themes";
	expect(queryByTestId(string)).toBeInTheDocument();
});

test("Should not render themes before time is 0", () => {
	const { queryByTestId } = render(<Themes />);
	const string = "senior-themes";
	expect(queryByTestId(string)).not.toBeInTheDocument();
});

test("Snapshot test", () => {
	const view = render(<Themes />);
	expect(view).toMatchSnapshot();
});
