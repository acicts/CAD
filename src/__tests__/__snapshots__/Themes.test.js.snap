// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="Container"
        id="themes"
      >
        <div
          class="Container undefined"
        >
          <h1>
            <span>
              #
            </span>
             
            <span
              style="color: rgb(7, 37, 58);"
            >
              Themes
            </span>
          </h1>
          <div
            class="TextContainer"
          >
            <div
              class="PendContainer"
            >
              <div
                class="Bar"
                style="background-color: rgb(7, 37, 58); height: 0vh; opacity: 0;"
              />
              <div
                class="Box"
                style="background-color: rgb(7, 37, 58);"
              />
            </div>
            <div
              class="children"
            >
              <div
                class="TimeContainer"
              >
                <h1>
                  Themes Will Be Revealed In!
                </h1>
                <div
                  class="TimeBar"
                >
                  <div>
                    <p>
                      0
                    </p>
                    <p>
                      Days
                    </p>
                  </div>
                  <div
                    class="Dots"
                  >
                    :
                  </div>
                  <div>
                    <p>
                      16
                    </p>
                    <p>
                      Hours
                    </p>
                  </div>
                  <div
                    class="Dots"
                  >
                    :
                  </div>
                  <div>
                    <p>
                      45
                    </p>
                    <p>
                      Mins
                    </p>
                  </div>
                  <div
                    class="Dots"
                  >
                    :
                  </div>
                  <div>
                    <p>
                      37
                    </p>
                    <p>
                      Seconds
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="Container"
      id="themes"
    >
      <div
        class="Container undefined"
      >
        <h1>
          <span>
            #
          </span>
           
          <span
            style="color: rgb(7, 37, 58);"
          >
            Themes
          </span>
        </h1>
        <div
          class="TextContainer"
        >
          <div
            class="PendContainer"
          >
            <div
              class="Bar"
              style="background-color: rgb(7, 37, 58); height: 0vh; opacity: 0;"
            />
            <div
              class="Box"
              style="background-color: rgb(7, 37, 58);"
            />
          </div>
          <div
            class="children"
          >
            <div
              class="TimeContainer"
            >
              <h1>
                Themes Will Be Revealed In!
              </h1>
              <div
                class="TimeBar"
              >
                <div>
                  <p>
                    0
                  </p>
                  <p>
                    Days
                  </p>
                </div>
                <div
                  class="Dots"
                >
                  :
                </div>
                <div>
                  <p>
                    16
                  </p>
                  <p>
                    Hours
                  </p>
                </div>
                <div
                  class="Dots"
                >
                  :
                </div>
                <div>
                  <p>
                    45
                  </p>
                  <p>
                    Mins
                  </p>
                </div>
                <div
                  class="Dots"
                >
                  :
                </div>
                <div>
                  <p>
                    37
                  </p>
                  <p>
                    Seconds
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
